src/components/SimplePriceChart/SimplePriceChart.tsx

```diff
@@ -69,19 +69,19 @@ const MarketDataChart: React.FC<SimplePriceChartProps> = ({
         setPrice(hourlyData?.slice(-30 * hourlyDataInterval)) //last 30 days, hourly
       }
     }, 0)
-  }, [durationSelector, data])
+  }, [durationSelector, data, hourlyData])
 
   const handleDailyButton = () => {
     setDurationSelector(0)
-    setIsDaily(true)
+    setChartRange(1)
   }
   const handleWeeklyButton = () => {
     setDurationSelector(1)
-    setIsDaily(false)
+    setChartRange(7)
   }
   const handleMonthlyButton = () => {
     setDurationSelector(2)
-    setIsDaily(false)
+    setChartRange(30)
```

Comment:
small knit: using an enum for these values might improve readibility

```
enum PriceChartRangeOption {
  DAILY_PRICE_RANGE = 1,
  WEEKLY_PRICE_RANGE = 7,
  MONTHLY_PRICE_RANGE = 30,
}

setChartRange(PriceChartRangeOption.DAILY_PRICE_RANGE)
```

Reply:
That's a great point, easy change to make as well.